src/blockchain_plugin.cpp

```diff
@@ -47,6 +47,35 @@ class blockchain_plugin_impl : std::enable_shared_from_this<blockchain_plugin_im
                      exec_engine->open();
                   }
 
+                  if (new_block->consensus_parameter_index) {
+                     if (*new_block->consensus_parameter_index == new_block->header.number) {
+                        // Parameters updated in this block.
+                        SILKWORM_ASSERT(new_block->consensus_parameters_cache.has_value());
+                        update_consensus_parameters(exec_engine->get_tx(), new_block->header.number, *new_block->consensus_parameters_cache);
+                        last_consensus_parameters = new_block->consensus_parameters_cache;
+                        last_consensus_parameters_index = new_block->consensus_parameter_index;
+                     }
+                     else if (last_consensus_parameters_index && *last_consensus_parameters_index == *new_block->consensus_parameter_index) {
+                        // Copy over parameters for last block.
+                        SILKWORM_ASSERT(last_consensus_parameters.has_value());
+                        new_block->consensus_parameters_cache = last_consensus_parameters;
+                     } else {
+                        // Parameter not cached, happen during startup or fork.
+                        new_block->consensus_parameters_cache = read_consensus_parameters(exec_engine->get_tx(), *new_block->consensus_parameter_index);
+                        // In such cases, the parameter MUST be in db.
+                        SILKWORM_ASSERT(new_block->consensus_parameters_cache.has_value());
+
+                        last_consensus_parameters_index = new_block->consensus_parameter_index;
+                        last_consensus_parameters = new_block->consensus_parameters_cache;
+                     }  
+                  }
+                  else {
+                     // Should only reach here if fork happens during version update.
+                     // TODO: Add guards according to eosevm versions.
+                     last_consensus_parameters_index = std::nullopt;
+                     last_consensus_parameters = std::nullopt;
+                  }
+                  
```

Comment:
We don't need this and the last_consensus_parameters_index / last_consensus_parameters variables